### PR TITLE
Replace utility commands with cbmc-viewer subcommands

### PR DESCRIPTION
**Pull request by markrtuttle**
This pull request replaces the utility commands like `make-trace` viewer subcommands like `cbmc-viewer trace`.

This pull request performs three transformations:
* The first six commits modify that make-* commands to invoke the make_* methods with the full dict args in preparation for the view subcommands setting args.func to the make_* method and invoking args.func(args).  In some cases this requires replacing the position argument like result.xml with an option argument like --result result.xml consistent with viewer.
* The next three commits modify the regression testing tool difference.py to test the utility commands make-* in addition to viewer itself.
* The final three commits implement the subcommands.  We now define one dict OPTIONS that describes all possible optional arguments, define one dict SUBPARSERS that describes the subcommands and the flags in OPTIONS they need, and pass this all to a single, uniform create_parser method.  One commit removes the make-* commands and one commit modifies difference.py to test the make-* functionality using subcommands.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: tests/bin/difference.py [hunk 1]
```diff
@@ -214,10 +217,112 @@ def update_jobs(jobs, new_viewer, new_report_root):
...
+
+def make_trace(cmd, opts, old):
+    cmd = Path(cmd).with_name('make-trace')
+    res = opts['--result']
+    src = opts['--srcdir']
+    if old:
+        return f"{cmd} {res} --srcdir {src}"
+    return f"{cmd} --result {res} --srcdir {src}"
+
+def make_kind(job, kind, viewer, report_root, old):
```
Comment (angelonakos) [inline tests/bin/difference.py:293]: Remove `old`?

**Review by angelonakos — APPROVED** [review]

File: tests/bin/difference.py (hunk 1 above)
Comment (SaswatPadhi) [reply tests/bin/difference.py:293]: `old` is being used in line 308 though.

File: tests/bin/difference.py [hunk 2]
```diff
@@ -225,41 +202,140 @@ def update_command(command, new_viewer, new_reportdir):
...
+    pairs = zip(keys, vals)
+
+    return command, dict(pairs)
+
+def make_coverage(cmd, opts, old):
+    cov = opts['--coverage']
+    src = opts['--srcdir']
+    if old:
+        cmd = Path(cmd).with_name('make-coverage')
+        return f"{cmd} {cov} --srcdir {src}"
```
Comment (SaswatPadhi) [inline tests/bin/difference.py:235]: This is within `tests/` so not a big deal for end users, but this function makes me nervous. It's very fragile -- expects exactly one argument (`value`) for each cmdline flag (`key`) and doesn't allow positional arguments. Would it be possible to reuse our parser here?

Also, it looks like `parsed_command` is being called to extract `old_opts` in `make_kind`, but many `make_*` functions generate positional arguments without a "key" that this function expects. For instance, the following `old` case in `make_coverage` generates a positional argument `{cov}` without a "key":
https://github.com/model-checking/cbmc-viewer/blob/90d3dbd29b02a410f51ef0f7d29536f7eb8c96fb/tests/bin/difference.py#L233-L235

Is this correct?

File: tests/bin/difference.py (hunk 1 above)
Comment (angelonakos) [reply tests/bin/difference.py:293]: Thanks for pointing out.

File: tests/bin/difference.py (hunk 2 above)
Comment (SaswatPadhi) [reply tests/bin/difference.py:235]: Sorry, just noticed that I posted this comment on the wrong line. I am talking about the function above, `parsed_command` in my comment.

**Pull request merged**